Building from a class not in a package doesn't work

Given the following code:

```java
package simple;

import static cc.quarkus.qcc.runtime.CNative.*;

public class Simple {
    @extern
    public static native int putchar(int arg);

    @export
    public static int main() {
        putchar('!');
        putchar('\n');
        return 0;
    }

    public static void main(String[] args) {
        // make driver happy
    }
}
```

I am able to create a binary executable and run it with:
```shell
javac -cp ~/.m2/repository/cc/quarkus/qcc-runtime-api/1.0.0-SNAPSHOT/qcc-runtime-api-1.0.0-SNAPSHOT.jar simple/Simple.java
jbang cc.quarkus:qcc-main:1.0.0-SNAPSHOT --boot-module-path ~/.m2/repository/cc/quarkus/qccrt-java.base/11.0.1-SNAPSHOT/qccrt-java.base-11.0.1-SNAPSHOT.jar:./ -o target simple.Simple
./target/a.out
```

However if I remove the first line, i.e., `package simple;` and try it with:
```shell
javac -cp ~/.m2/repository/cc/quarkus/qcc-runtime-api/1.0.0-SNAPSHOT/qcc-runtime-api-1.0.0-SNAPSHOT.jar simple/Simple.java
jbang cc.quarkus:qcc-main:1.0.0-SNAPSHOT --boot-module-path ~/.m2/repository/cc/quarkus/qccrt-java.base/11.0.1-SNAPSHOT/qccrt-java.base-11.0.1-SNAPSHOT.jar:./simple -o target Simple
```

I am getting:
```
[jbang] Resolving dependencies...
[jbang]     Resolving cc.quarkus:qcc-main:1.0.0-SNAPSHOT...Done
[jbang] Dependencies resolved
debug: [IntrinsicBasicBlockBuilder] found InstanceValueIntrisic for owner(Ljava/lang/Object;) name(getClass) descriptor(()Ljava/lang/Class;)
Simple.java:11: warning: Invalid node encountered (cannot directly lower new)
    member: method main@ bci 2
  location: type Simple
Simple.java:11: warning: Invalid node encountered (cannot directly lower throw)
    member: method main@ bci 2
  location: type Simple
Throwable.java:255: warning: Virtual invocation not supported yet
    member: constructor <init>@ bci 24
  location: type Throwable
Throwable.java:115: warning: Virtual invocation not supported yet
  location: type Throwable
Throwable.java:160: warning: Invalid node encountered (cannot directly lower new)
  location: type Throwable
Throwable.java:1068: warning: Invalid node encountered (cannot directly lower new)
  location: type Throwable
Throwable.java: warning: Invalid node encountered (cannot directly lower monitorEnter)
    member: method fillInStackTrace
  location: type Throwable
Throwable.java:787: warning: Virtual invocation not supported yet
    member: method fillInStackTrace@ bci 16
  location: type Throwable
Throwable.java:785: warning: Invalid node encountered (cannot directly lower throw)
    member: method fillInStackTrace@ bci 1
  location: type Throwable
Throwable.java:790: warning: Invalid node encountered (cannot directly lower monitorExit)
    member: method fillInStackTrace@ bci 28
  location: type Throwable
Class.java:3473: warning: Virtual invocation not supported yet
    member: method desiredAssertionStatus@ bci 1
  location: type Class
Class.java:3480: warning: Invalid node encountered (cannot directly lower new)
    member: method desiredAssertionStatus@ bci 20
  location: type Class
Class.java:3480: warning: Invalid node encountered (cannot directly lower monitorEnter)
    member: method desiredAssertionStatus@ bci 20
  location: type Class
Class.java:3480: warning: Invalid node encountered (cannot directly lower throw)
    member: method desiredAssertionStatus@ bci 20
  location: type Class
Class.java:3484: warning: Invalid node encountered (cannot directly lower new)
    member: method desiredAssertionStatus@ bci 40
  location: type Class
Class.java:3484: warning: Invalid node encountered (cannot directly lower throw)
    member: method desiredAssertionStatus@ bci 40
  location: type Class
Class.java:3484: warning: Invalid node encountered (cannot directly lower monitorExit)
    member: method desiredAssertionStatus@ bci 40
  location: type Class
Class.java:3482: warning: Virtual invocation not supported yet
    member: method desiredAssertionStatus@ bci 30
  location: type Class
Class.java:3482: warning: Virtual invocation not supported yet
    member: method desiredAssertionStatus@ bci 33
  location: type Class
Class.java:3480: warning: Invalid node encountered (cannot directly lower throw)
    member: method desiredAssertionStatus@ bci 20
  location: type Class
Class.java:3482: warning: Invalid node encountered (cannot directly lower new)
    member: method desiredAssertionStatus@ bci 37
  location: type Class
Class.java:3482: warning: Invalid node encountered (cannot directly lower throw)
    member: method desiredAssertionStatus@ bci 37
  location: type Class
Class.java:3482: warning: Invalid node encountered (cannot directly lower monitorExit)
    member: method desiredAssertionStatus@ bci 37
  location: type Class
Class.java:3287: warning: Invalid node encountered (cannot directly lower new)
  location: type Class
Class.java:3444: warning: Invalid node encountered (cannot directly lower new)
  location: type Class
Class.java:793: warning: Virtual invocation not supported yet
    member: method getName@ bci 14
  location: type Class
ClassLoader.java:2930: warning: Invalid node encountered (cannot directly lower new)
    member: method desiredAssertionStatus@ bci 6
  location: type ClassLoader
ClassLoader.java:2930: warning: Invalid node encountered (cannot directly lower monitorEnter)
    member: method desiredAssertionStatus@ bci 6
  location: type ClassLoader
ClassLoader.java:2930: warning: Invalid node encountered (cannot directly lower throw)
    member: method desiredAssertionStatus@ bci 6
  location: type ClassLoader
ClassLoader.java:2935: warning: Invalid node encountered (cannot directly lower new)
    member: method desiredAssertionStatus@ bci 12
  location: type ClassLoader
ClassLoader.java:2935: warning: Virtual invocation not supported yet
    member: method desiredAssertionStatus@ bci 12
  location: type ClassLoader
ClassLoader.java:2935: warning: Invalid node encountered (cannot directly lower throw)
    member: method desiredAssertionStatus@ bci 12
  location: type ClassLoader
ClassLoader.java:2930: warning: Invalid node encountered (cannot directly lower throw)
    member: method desiredAssertionStatus@ bci 6
  location: type ClassLoader
ClassLoader.java:2937: warning: Virtual invocation not supported yet
    member: method desiredAssertionStatus@ bci 26
  location: type ClassLoader
ClassLoader.java:2937: warning: Invalid node encountered (cannot directly lower new)
    member: method desiredAssertionStatus@ bci 30
  location: type ClassLoader
ClassLoader.java:2937: warning: Invalid node encountered (cannot directly lower throw)
    member: method desiredAssertionStatus@ bci 30
  location: type ClassLoader
ClassLoader.java:2937: warning: Invalid node encountered (cannot directly lower monitorExit)
    member: method desiredAssertionStatus@ bci 30
  location: type ClassLoader
Collections.java:4301: warning: Invalid node encountered (cannot directly lower new)
  location: type Collections
Collections.java:4425: warning: Invalid node encountered (cannot directly lower new)
  location: type Collections
Collections.java:4527: warning: Invalid node encountered (cannot directly lower new)
  location: type Collections
ClassLoader.java:233: warning: Virtual invocation not supported yet
  location: type ClassLoader
ClassLoader.java:309: warning: Invalid node encountered (cannot directly lower new)
  location: type ClassLoader
ClassLoader.java:2719: warning: Invalid node encountered (cannot directly lower new)
  location: type ClassLoader
HashSet.java:106: warning: Invalid node encountered (cannot directly lower new)
    member: constructor <init>@ bci 5
  location: type HashSet
Boolean.java:52: warning: Invalid node encountered (cannot directly lower new)
  location: type Boolean
Boolean.java:58: warning: Invalid node encountered (cannot directly lower new)
  location: type Boolean
HashSet.java:99: warning: Invalid node encountered (cannot directly lower new)
  location: type HashSet
error: Function `exact.java.util.Map.get.ref.class.java-lang-Object.2.ref.interface.java-util-Map.ref.class.java-lang-Object` has no body
Boolean.java:52: error: llvm: Unrecognized value class cc.quarkus.qcc.graph.New
  location: type Boolean
Boolean.java:58: error: llvm: Unrecognized value class cc.quarkus.qcc.graph.New
  location: type Boolean
error: Pre-generate hook failed: java.lang.NullPointerException
Location.java:268: note: This is the location of the exception
    member: method setElement
  location: type Location$Builder
Compilation failed with 4 error(s) and 47 warning(s)
```

--- a/compiler/src/main/java/cc/quarkus/qcc/type/definition/element/MethodElement.java
+++ b/compiler/src/main/java/cc/quarkus/qcc/type/definition/element/MethodElement.java
@@ -30,7 +30,11 @@ public final class MethodElement extends InvokableElement implements NamedElemen
     }
 
     public String toString() {
-        return getEnclosingType().getDescriptor().getPackageName()+"."+getEnclosingType().getDescriptor().getClassName()+"."+getName()+getDescriptor();
+        final String packageName = getEnclosingType().getDescriptor().getPackageName();
+        if (packageName.isEmpty()) {
+            return getEnclosingType().getDescriptor().getClassName()+"."+getName()+getDescriptor();
+        }
+        return packageName+"."+getEnclosingType().getDescriptor().getClassName()+"."+getName()+getDescriptor();
     }
 
     public String getName() {

--- a/compiler/src/main/java/cc/quarkus/qcc/type/generic/NestedClassTypeSignature.java
+++ b/compiler/src/main/java/cc/quarkus/qcc/type/generic/NestedClassTypeSignature.java
@@ -63,6 +63,12 @@ public final class NestedClassTypeSignature extends ClassTypeSignature {
 
     ClassTypeDescriptor makeDescriptor(final ClassContext classContext) {
         ClassTypeDescriptor encDesc = enclosing.asDescriptor(classContext);
-        return ClassTypeDescriptor.synthesize(classContext, encDesc.getPackageName() + '/' + encDesc.getClassName() + '$' + getIdentifier());
+        final String internalName;
+        if (encDesc.getPackageName().isEmpty()) {
+            internalName = encDesc.getClassName() + '$' + getIdentifier();
+        } else {
+            internalName = encDesc.getPackageName() + '/' + encDesc.getClassName() + '$' + getIdentifier();
+        }
+        return ClassTypeDescriptor.synthesize(classContext, internalName);
     }
 }

--- a/plugins/native/src/main/java/cc/quarkus/qcc/plugin/native_/FunctionTypeResolver.java
+++ b/plugins/native/src/main/java/cc/quarkus/qcc/plugin/native_/FunctionTypeResolver.java
@@ -76,7 +76,12 @@ public class FunctionTypeResolver implements DescriptorTypeResolver.Delegating {
                             TypeDescriptor pointeeDesc = pointeeSig.asDescriptor(classCtxt);
                             if (pointeeDesc instanceof ClassTypeDescriptor) {
                                 ClassTypeDescriptor classDesc = (ClassTypeDescriptor) pointeeDesc;
-                                String name = classDesc.getPackageName() + '/' + classDesc.getClassName();
+                                final String name;
+                                if (classDesc.getPackageName().isEmpty()) {
+                                    name = classDesc.getClassName();
+                                } else {
+                                    name = classDesc.getPackageName() + '/' + classDesc.getClassName();
+                                }
                                 if (name.equals("java/lang/Object")) {
                                     // special case: it's really an "any" pointer with extra front-end guards on it
                                     return ts.getVoidType().asConst().getPointer();

--- a/plugins/verification/src/main/java/cc/quarkus/qcc/plugin/verification/ClassLoadingBasicBlockBuilder.java
+++ b/plugins/verification/src/main/java/cc/quarkus/qcc/plugin/verification/ClassLoadingBasicBlockBuilder.java
@@ -181,7 +181,13 @@ public class ClassLoadingBasicBlockBuilder extends DelegatingBasicBlockBuilder {
     }
 
     private boolean loadClass(ClassTypeDescriptor desc) {
-        DefinedTypeDefinition definedType = getClassContext().findDefinedType(desc.getPackageName() + "/" + desc.getClassName());
+        final String typeName;
+        if (desc.getPackageName().isEmpty()) {
+            typeName = desc.getClassName();
+        } else {
+            typeName = desc.getPackageName() + "/" + desc.getClassName();
+        }
+        DefinedTypeDefinition definedType = getClassContext().findDefinedType(typeName);
         if (definedType == null) {
             return false;
         }

--- a/plugins/verification/src/main/java/cc/quarkus/qcc/plugin/verification/MemberResolvingBasicBlockBuilder.java
+++ b/plugins/verification/src/main/java/cc/quarkus/qcc/plugin/verification/MemberResolvingBasicBlockBuilder.java
@@ -237,7 +237,13 @@ public class MemberResolvingBasicBlockBuilder extends DelegatingBasicBlockBuilde
     }
 
     private DefinedTypeDefinition resolveDescriptor(final ClassTypeDescriptor owner) {
-        return getClassContext().findDefinedType(owner.getPackageName() + "/" + owner.getClassName());
+        final String typeName;
+        if (owner.getPackageName().isEmpty()) {
+            typeName = owner.getClassName();
+        } else {
+            typeName = owner.getPackageName() + "/" + owner.getClassName();
+        }
+        return getClassContext().findDefinedType(typeName);
     }
 
     private void nsfe() {

